### DISCUSSION TITLE
SYS: stabilizer [FAIL] and android app con't control fly

### DISCUSSION THREAD

**Discussion by justinleeyang** [bitcraze/discussions#1575] [Q&A]
crazyflie 2.1  use BMI088 and BMP388, sensor pass, but Android app con't control fly, why ? 
```
SYS: Crazyflie 2.0 is up and running!
SYS: Build 32:9343aa686600 (2023.07 +32) MODIFIED
SYS: I am 0x3836323634365104001C002B and I have 1024KB of flash!
CFGBLK: v1, verification [OK]
DECK_CORE: 0 deck(s) found
SENSORS: Forcing sensors to SensorImplementation_bmi088_bmp388
IMU: BMI088: Using I2C interface.
IMU: BMI088 Gyro connection [OK].
IMU: BMI088 Accel connection [OK]
IMU: BMP388 I2C connection [OK]
ESTIMATOR: Using Complementary (1) estimator
CONTROLLER: Using PID (1) controller
MTR-DRV: Using brushed motor driver
SYS: About to run tests in system.c.
SYS: NRF51 version: 2023.02 +1*
EEPROM: I2C connection [OK].
STORAGE: Storage check [OK].
IMU: BMI088 gyro returning x=0 y=0 z=0 [FAILED]
IMU: BMI088 gyro self-test [OK]
SYS: stabilizer [FAIL]
```

**Comment by justinleeyang** [comment]
IMU BMI088 Gyro and Accel connection [OK]，but why gyro returning x=0 y=0 z=0 [FAILED] ？will result in stabilizer [FAIL] ?

**Reply by gemenerik** [reply]
It doesn't seem to be an issue with the app.

Being connected to the gyro doesn’t necessarily mean it’s providing the correct data. It’s failing an initialization test, which prevents the stabilizer from functioning—making it impossible to fly.

Could you try reflashing the latest firmware release?

**Reply by justinleeyang** [reply]
the same code, other crazyflie2.1's gyro is OK, only one fail

**Reply by gemenerik** [reply]
I just noticed the line `SENSORS: Forcing sensors to SensorImplementation_bmi088_bmp388`. Why are you forcing this? Please try stock release firmware. If that does not help, it may be that your gyro is broken. Please send an e-mail to support@bitcraze.io

**Reply by justinleeyang** [reply]
remove force, the problem still exist. the sensor maybe is broken.
